### PR TITLE
feat(colormode): add dark theme initial support

### LINKED ISSUE #30: Add dark mode to site
This project uses chakra ui which has docs on adding dark mode here
[https://chakra-ui.com/color-mode](https://chakra-ui.com/color-mode)

Goal: Add dark mode to the site with a button in the header to toggle. The button can just be a plain button for now and would be added to Header.tsx. There already is a theme.js where the dark mode colors can start.

### PR DESCRIPTION
**Pull request by luisFilipePT**
## Changes

Add darkmode support

*Notes*
* Footer is "light" in both modes, the dark didn't quite seem right, but if you want tell me and I will open a new pr.
* Added modes to the colors in the `theme`, right now just switched primary and secondary on dark because of the active tab, this is useful for toggling the color mode in components with explicit colors.

## Screens

Dark
image

Light
image

closes #30

Cheers 👐

### CODE REVIEW

File: front/src/components/header/subcomponents/ToggleColorModeButton.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+const ToggleColorModeButton = () => {
+  const { colorMode, toggleColorMode } = useColorMode();
+
+  return (
+    <IconButton
+      aria-label="toggle color mode"
+      ml=" 12px"
+      onClick={toggleColorMode}
+      icon={colorMode === 'light' ? 'moon' : 'sun'}
```
Comment (trybick) [inline front/src/components/header/subcomponents/ToggleColorModeButton.tsx]: nit - alphabetize props names

**Review by trybick — COMMENTED** [review]
@luisFilipePT It looks great :)  Nice job on the header. Would it be possible to add the dark colors to the footer too?

**Comment by luisFilipePT** [comment]
> @luisFilipePT It looks great :)  Nice job on the header. Would it be possible to add the dark colors to the footer too?

Thanks ☺️. Sure, you can merge that one and probably today I can open a new PR for the footer, just to avoid conflicts if there's more merges in the meantime.

File: front/src/components/header/subcomponents/ToggleColorModeButton.tsx (hunk 1 above)
Comment (luisFilipePT) [reply front/src/components/header/subcomponents/ToggleColorModeButton.tsx]: done https://github.com/trybick/tv-minder/pull/32/commits/b1914d8d6ea08ac6a748bdd8726afb966e056563

Comment (luisFilipePT) [reply front/src/components/header/subcomponents/ToggleColorModeButton.tsx]: @trybick if you want I can add https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-sort-props.md to enforce it everywhere 😄

Comment (trybick) [reply front/src/components/header/subcomponents/ToggleColorModeButton.tsx]: Thank you. Yes please let's enforce it everywhere 🚀 . that would be great!

**Comment by trybick** [comment]
@luisFilipePT Sounds good with a later PR for the footer. Great work 🎉

**Comment by trybick** [comment]
@luisFilipePT Could you also add the toggle button to the header when logged in? Right now it's only showing when logged out. 

It should be able to added to this ternary or put outside the ternary
https://github.com/trybick/tv-minder/blob/master/front/src/components/header/Header.tsx#L150

File: front/src/components/header/subcomponents/ToggleColorModeButton.tsx (hunk 1 above)
Comment (trybick) [reply front/src/components/header/subcomponents/ToggleColorModeButton.tsx]: @luisFilipePT Did you still want to make a PR for this eslint rule?

Comment (luisFilipePT) [reply front/src/components/header/subcomponents/ToggleColorModeButton.tsx]: Yes, I can do it this afternoon 😄

Comment (luisFilipePT) [reply front/src/components/header/subcomponents/ToggleColorModeButton.tsx]: @trybick btw I am also looking at the color function and one question, does the color needs to be static (defined at the beginning) or can it be generated? as long as it is always the same color for the same id?

Comment (trybick) [reply front/src/components/header/subcomponents/ToggleColorModeButton.tsx]: great! 

for the random color for the shows on the calendar? Hmm, it generates the color based on the ID, so I think every time the same ID will have the same color.  

The `return index + 1` was just a hack to get it to work if there are two IDs with the same index but I would like to get it working so if there are 3 or more shows with the same index they will all have a unique index.

**Pull request merged**